### DISCUSSION TITLE
export type PostgrestError from '@supabase/supabase-js'

### DISCUSSION THREAD

**Discussion by d-mok** [supabase/supabase#2015] [Feature Requests]
I am using typescript. I want to write a function called `handleError` to handle the `data` and `error` objects returned by `.select()`.
```
let { data, error } = await supabase.from<T>('myTable').select('*')
handleError(data, error)
```
But when I defined my `handleError` function:
```
import { PostgresqlError } from '@supabase/supabase-js';

function handleError<T>(data: T[] | null, error: PostgresqlError) {
   // some logic here
}
```
it says:
`'"@supabase/supabase-js"' has no exported member 'PostgresqlError'`

I am forced to use `unknown` or `any` type for the error parameter, which is no good.
Can supabase export the type `PostgresqlError`? Or is there a better practice for my purpose?

**Comment by ghost** [comment]
They probably haven't exported the type for `error` which is `PostgrestError | null`.

https://github.com/supabase/postgrest-js/blob/915df5d548d6bf53aa8e21b731600dccee700113/src/lib/types.ts#L8-L13

**Comment by soedirgo** [comment]
Hey @d-mok, `PostgrestError` is exported now in supabase-js v1.16.1. Let me know if that works for you :)

**Reply by d-mok** [reply]
yeah, thanks!
